src/PostgREST/Error.hs

```diff
@@ -176,10 +176,10 @@ instance JSON.ToJSON ApiRequestError where
     (Just $ JSON.String $ "'" <> origin <> "' and '" <> target <> "' do not form a many-to-one or one-to-one relationship")
     Nothing
 
-  toJSON (SpreadNotToOne origin target) = toJsonPgrstError
+  toJSON SpreadToManyNoSelectAll = toJsonPgrstError
     ApiRequestErrorCode19
-    ("A spread operation on '" <> target <> "' is not possible")
-    (Just $ JSON.String $ "'" <> origin <> "' and '" <> target <> "' do not form a many-to-one or one-to-one relationship")
+    "Selecting '*' is not allowed in a spread operation for one-to-many or many-to-many relationships"
```

Comment:
I think that, instead of forbidding `*`, it would be better to expand the selection using the column names from the Schema Cache. In the same way as aggregates do it here:

https://github.com/PostgREST/postgrest/blob/c3070bbd4ecd42d02fc3a4ec00cdd2fbdf2aa224/src/PostgREST/Plan.hs#L410-L414